src/components/Button/BaseButton.tsx

```diff
@@ -150,26 +151,30 @@ export const BaseButton: FC<InternalButtonProps> = React.forwardRef(
         );
 
         const getButtonText = (
-            buttonTextClassNames: string,
+            buttonTextClasses: ArgumentArray,
             text: string
         ): JSX.Element => (
-            <span className={buttonTextClassNames}>
+            <Atom of="span" classes={buttonTextClasses}>
                 {text ? text : 'Button'}
-            </span>
+            </Atom>
         );
 
         return (
             <>
-                <button
+                <Atom<
+                    React.ButtonHTMLAttributes<HTMLButtonElement>,
+                    HTMLButtonElement
+                >
                     {...rest}
+                    of="div"
```

Comment:
The amount of code we need to write increases a lot cause of this change for eg:
`<button>` vs `<Atom<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement> of="button">`

This will both make it cumbersome to use and will effet overall code readability. Also everytime want to now use a div or any element as an atom there will be uncessary js running in the background. 
I think we need to step back and rethink this approach.

Reply:
@ychhabra-eightfold Yes, there is a level of complexity involved here. 
The tradeoff is to have fine control of what can be passed to a 'div'. Also, atoms are only for primitive components and used internally within the library.